src/android/com/perfectco/cordova/TlsPskServer.java

```diff
@@ -40,6 +41,7 @@ public void start(int port, final ExecutorService executor, final AcceptCallback
     }
 
     socket = new ServerSocket(port);
+    socket.setSoTimeout(SOCKET_TIMEOUT_MS);
```

Comment:
Setting the timeout here only applies to the `accept()`, not to the actual communication socket. Set this on the result of `accept()` below instead.